pyiron/atomistics/structure/analyse.py

```diff
@@ -0,0 +1,72 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+import numpy as np
+from pyiron_base import Settings
+from sklearn.cluster import AgglomerativeClustering
+
+__author__ = "Joerg Neugebauer, Sam Waseda"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "1.0"
+__maintainer__ = "Sam Waseda"
+__email__ = "waseda@mpie.de"
+__status__ = "production"
+__date__ = "Sep 1, 2017"
+
+s = Settings()
+
+class Analyse:
+    """ Class to analyse atom structure.  """
+    def __init__(self, ref_structure):
+        """
+        Args:
+            ref_structure (pyiron.atomistics.structure.atoms.Atoms): reference Atom structure
+        """
+        self._ref_structure = ref_structure
+
+    def __repr__(self):
+        """
+        Returns: __repr__
+        """
+        return ''
+
+    def get_layers(self, distance_threshold=0.01, id_list=None):
+        """
+        Get an array of layer numbers.
```

Comment:
Do I understand correctly that this is somehow scanning linkages in the cartesian x,y,z directions?

I think the docstring still needs more fleshing out. 

Also, I don't understand it well enough so maybe it takes care of this already, but what about non-rectangular cells?

Finally, does there need to be some sort of option to use the unwrapped positions, or even to force-wrap the positions?

Reply:
> Do I understand correctly that this is somehow scanning linkages in the cartesian x,y,z directions?

Yes!

> 
> I think the docstring still needs more fleshing out.
> 
> Also, I don't understand it well enough so maybe it takes care of this already, but what about non-rectangular cells?

I also thought about it, but for now it simply looks at the absolute coordinates. I can also include something like `frame` which would take a 2d tensor to define the planes.

> 
> Finally, does there need to be some sort of option to use the unwrapped positions, or even to force-wrap the positions?

You have to have wrapped the positions beforehand. Do you want me to add an extra argument for that?

Reply:
> I also thought about it, but for now it simply looks at the absolute coordinates. I can also include something like frame which would take a 2d tensor to define the planes.

Ok, I guess this should be clarified in the docstring but I don't have any problem with it as current behaviour.

> You have to have wrapped the positions beforehand. Do you want me to add an extra argument for that?

Hmm, no, maybe no requirement to wrap beforehand since, as you say, you can do that already at the structure level. But does allowing a flag to use unwrapped coordinates make sense?

Reply:
I added wrapping

Reply:
> Also, I don't understand it well enough so maybe it takes care of this already, but what about non-rectangular cells?
> 

In the end I added `planes` where you can say things like `planes=[1,1,1]` to get planes along the `[1,1,1]`-direction.